### PR TITLE
clean up tests to use FactCheck

### PR DESCRIPTION
**Pull request by davidlizeng**
Reorganized test.jl and test2.jl into affine, lp, and socp test files as per #42 . 
All tests also now use the FactCheck package.

### CODE REVIEW

**Comment by karanveerm** [comment]
Few issues:

- This will always pass travis because of the exit status. Read: https://github.com/JuliaLang/FactCheck.jl#exit-status and fix.
- It's preferable to have test_lp separate, so we can test other LP solvers on it as well.

**Comment by karanveerm** [comment]
@davidlizeng I think this is ready to merge into master. Coverage should go up another 20% after we test with SCS on travis, after which most of the uncovered lines are error checks, etc. of lower priority.

File: REQUIRE [hunk 1]
```diff
@@ -1,3 +1,4 @@
 julia 0.3
 MathProgBase 0.3.8
 ECOS 0.4.1
+FactCheck 0.2.5
```
Comment (mlubin) [inline REQUIRE]: This is usually listed in ``test/REQUIRE`` since it's only needed to run tests.

Comment (karanveerm) [inline REQUIRE]: What would be the nicest way for FactCheck to still be present for travis?
JuMP seems to be doing this: https://github.com/JuliaOpt/JuMP.jl/blob/master/.travis.yml#L11

Comment (mlubin) [inline REQUIRE]: Oops, I guess we never tidied that up. @IainNZ 
If you just use ``Pkg.test("Convex", coverage=true)`` then first the testing dependencies will be installed and then ``test/runtests.jl`` will be run. That's the "right" way to do it.

Comment (karanveerm) [inline REQUIRE]: Done, thanks! I'm going to keep this PR open for another few days to see if anyone has other feedback, and merge it then.

**Pull request merged**